### ISSUE TITLE
how to configure sidecar proxy to advertise "h2" as ALPN

### ISSUE THREAD

**Issue by rgf2004** [hashicorp/consul#12106]
#### The question:
How to configure the sidecar proxy to advertise `h2` as the ALPN?

#### The Scenario:
I have a gRPC service with a sidecar proxy. To it, other services should communicate natively. 

#### Steps 
- I followed the steps mentioned [here](https://learn.hashicorp.com/tutorials/consul/service-mesh-with-envoy-proxy?in=consul/getting-started) to register a service, and its sidecar proxy.
- I followed the steps mentioned [here](https://www.consul.io/docs/connect/native) to connect natively. 
- When other services attempt to establish a secure connection with this service's sidecar proxy, it fails with `Failed ALPN negotiation: Unable to find compatible protocol`. this is because the sidecar proxy doesn't expose the ALPN protocol, I assume so.  
- I see that envoy has a configuration for `alpn_protocols` [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/tls.proto) but I don't know if consul supports such configuration. 

Service definition, `grpc-service.hcl`
```
service {
  name = "grpc-service"
  id = "grpc-service-1"
  port = 9000

  connect = {
    sidecar_service = {}
  }

  check {
    id       = "grpc-service-check"
    grpc =  "127.0.0.1:9000"
    grpc_use_tls = false
    interval =  "10s"
  }
}
``` 
and this one, `additional-config.hcl`
```
Kind = "service-defaults"
Name = "grpc-service"
Protocol = "grpc"
MeshGateway {
	Mode = "none"
}
Meta { 
}
```

These are the used commands:
```
consul agent -dev -node machine
```
```
consul services register grpc-service.hcl 
```
```
consul config write additional-config.hcl
```
```
consul connect envoy -sidecar-for grpc-service-1 
```

**Comment by Amier3** [comment]
This is related to #11907 and should be fixed when the PR  #12170 is merged

**Comment by jcantwell-JC** [comment]
We are sad to see that PR https://github.com/hashicorp/consul/pull/12170 was closed. We have an Ingress Gateway configured through Nomad that needs to support `grpc`. However, without the envoy gateway broadcasting the appropriate `alpn` header, `grpc` clients will not connect to it (i.e. `python` client in this case).

```
Cannot check peer: missing selected ALPN property.
```

We were able to get this working using the changes in PR https://github.com/hashicorp/consul/pull/12170 **AND** also adding the following config to Nomad:
```
service {
  connect {
    gateway {
      proxy {
        config {
          protocol = "grpc"
        }
      }
   }
}
```
